libbeat/cmd/instance/beat.go

```diff
@@ -889,6 +889,11 @@ func (b *Beat) setupMonitoring(settings Settings) (report.Reporter, error) {
 	}
 
 	if monitoring.IsEnabled(monitoringCfg) {
+		err := monitoring.OverrideWithCloudSettings(monitoringCfg)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
If possible I'd prefer to not overwrite settings in a `*common.Config`. It's an anti-pattern we unfortunately use throughout Beats.

e.g. based on the CloudID structure one could introduce an interface type like:

```
type StackDetails interface {
  Username() string
  Password() string
  ElasticsearchURL() string
}
```

Then it's a matter of constructing the right object, instead of overwriting a shared configuration object that might already be applied partially.

I think for this PR it is ok as done, as monitoring settings are quite 'local'. But in general when dealing with configs we should prefer construction over modifications (treat *common.Config as readonly). It is a common issue that we sometimes did unpack a configuration before it has been modified.

Reply:
++ to not overwriting settings, I was only doing it to be consistent with how we are handling the global `cloud.id` and `cloud.auth` settings.